cov_profiles.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+#
+# Fetches coverage data for each fuzz test, combines them, and produces a
+# coverage profile that can be analyzed.
+if [[ "$#" -ne 1 ]]; then
+    echo "Usage: $0 LND_DIR"
+    exit 1
+fi
+
+readonly LND_DIR=$1
+readonly CACHE_DIR=$(mktemp -d)
+
+# Get the directory of cov_profiles.sh in case it is being called not in
+# lnd-fuzz.
+readonly BASE_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+readonly packages=("lnwire" "brontide" "htlcswitch/hop" "tlv" "watchtower/wtwire" "watchtower/wtclient" "zpay32") 
+
+# collect_combine_cov_profiles collects coverage for each fuzzing package and
+# then combines them.
+function collect_combine_cov_profiles {
+    local coverage_dirs=()
+
+    # Collect coverage profiles.
+    for p in ${packages[@]}; do
+        cd "${BASE_DIR}/${p}/testdata/fuzz"
+
+        for f in $(ls $PWD); do
+            # Move corpus to CACHE_DIR.
+            mkdir -p "${CACHE_DIR}/${f}"
+            cp -a "${BASE_DIR}/${p}/testdata/fuzz/${f}"/ "${CACHE_DIR}/${f}/"
+            num_inputs=$(ls "${CACHE_DIR}/${f}/" | wc -l | xargs)
```

Comment:
Is `xargs` needed here? It's just used for trimming whitespace, right?

Also, `ls` loads the full file list into memory, which might proof inefficient for large directories. Consider using this:


```suggestion
            num_inputs=$(find "${CACHE_DIR}/${f}/" -maxdepth 1 -type f | wc -l | tr -d '[:space:]')
```